frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -161,30 +160,27 @@ public void setTestResponse(EntityModel<Response> testResponse) {
     public UploadImageModel(final Guid limitToStorageDomainId, final DiskImage resumeUploadDisk) {
         if (resumeUploadDisk == null) {
             setDiskModel(new NewDiskModel() {
+                private final boolean isChangeable = limitToStorageDomainId == null;
+
                 @Override
                 public void initialize() {
                     super.initialize();
 
-                    getStorageDomain().setIsChangeable(limitToStorageDomainId == null);
-                    getDataCenter().setIsChangeable(limitToStorageDomainId == null);
+                    getStorageDomain().setIsChangeable(isChangeable);
+                    getDataCenter().setIsChangeable(isChangeable);
                     getStorageType().setIsChangeable(false);
                     getSize().setIsChangeable(false);
                 }
 
                 @Override
                 protected void updateStorageDomains(final StoragePool datacenter) {
-                    if (limitToStorageDomainId == null) {
+                    if (isChangeable) {
                         super.updateStorageDomains(datacenter);
```

Comment:
Skipping the call to `super.updateStorageDomains()` will leave the SD list uninitialized. The model behind SD list allows an illegal situation where you can select an item from a list that does not exist (it's not even  empty). This worked before (at least in most cases) however any other code may reach out to `getStorageDomain().getItems()` and then bad things can happen.

Reply:
so can we just set the items in `initStorageDomain` to a singleton list of the domain when it's not changeable?

Reply:
This could work. However we are bypassing the inherited flow there. I would rather rely on established extension points like filtering methods (which essentially will have the same effect - produce a singleton storage domain list and a singleton data center list).

Reply:
there's a room for improvement but we don't have the capacity to refactor this code and we still want to address the bug this PR is supposed to fix..